### ISSUE TITLE
Is Boomerang Output Compilable

### ISSUE THREAD

**Issue by humanitiesclinic** [BoomerangDecompiler/boomerang#223]

**Comment by humanitiesclinic** [comment]
Any suggestions?

**Comment by ceeac** [comment]
No, the output of Boomerang is generally not compilable, except for some trivial corner cases. You can have a look at the output of Boomerang for some simple programs in the `tests/regression-tests/expected-outputs/` directory. Think of it as pseudo-code that happens to be close to C (for some definition of "close").

As for correctness, this is currently not extensively tested. The best way to approach this is using a disassembly viewer and the decompiler output side-by-side and refactoring the decompiler output into compilable C while watching out for semantic differences between disassembly and decompiler output. Boomerang currently does not have a disassembly viewer so you have to use external tools for that. In any case, lots of manual labour will be required.
